src/util/sock.h

```diff
@@ -135,13 +136,16 @@ class Sock
      * @param[in] terminator Character up to which to read from the socket.
      * @param[in] timeout Timeout for the entire operation.
      * @param[in] interrupt If this is signaled then the operation is canceled.
+     * @param[in] max_data The maximum amount of data (in bytes) to receive without
+     * a terminator before giving up. Use `std::nullopt` to avoid this limit.
      * @return The data that has been read, without the terminating character.
      * @throws std::runtime_error if the operation cannot be completed. In this case some bytes may
      * have been consumed from the socket.
      */
     virtual std::string RecvUntilTerminator(uint8_t terminator,
                                             std::chrono::milliseconds timeout,
-                                            CThreadInterrupt& interrupt) const;
+                                            CThreadInterrupt& interrupt,
+                                            std::optional<size_t> max_data = std::nullopt) const;
```

Comment:
not sure what is more idiomatic
```suggestion
                                            std::optional<size_t> max_data = {}) const;
```

Reply:
personally I think nullopt is more readable. E.g., even though we know that {} is nullopt here, size_t x{} would be x == 0 right? So then could max_data={{}} be some(0)?

Reply:
yes, `std::nullopt` has the advantage of being more precise